# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/03
# default paths for config, models, schedules, and examples in site-packages/rfdiffusion, $RFDIFFUSION_MODELS, and $RFDIFFUSION_SCHEDULES
diff -ru RFdiffusion-1.1.0/rfdiffusion/inference/model_runners.py RFdiffusion-1.1.0_data_paths/rfdiffusion/inference/model_runners.py
--- RFdiffusion-1.1.0/rfdiffusion/inference/model_runners.py	2023-04-03 23:33:05.000000000 +0200
+++ RFdiffusion-1.1.0_data_paths/rfdiffusion/inference/model_runners.py	2023-08-08 18:04:13.224021765 +0200
@@ -17,7 +17,8 @@
 
 from rfdiffusion.model_input_logger import pickle_function_call
 import sys
-
+from rfdiffusion import __path__ as rfdiffusion_dir
+ 
 SCRIPT_DIR=os.path.dirname(os.path.realpath(__file__))
 
 TOR_INDICES  = util.torsion_indices
@@ -63,7 +64,7 @@
         if conf.inference.model_directory_path is not None:
             model_directory = conf.inference.model_directory_path
         else:
-            model_directory = f"{SCRIPT_DIR}/../../models"
+            model_directory = os.getenv('RFDIFFUSION_MODELS') 
 
         print(f"Reading models from {model_directory}")
 
@@ -122,7 +123,7 @@
         if conf.inference.schedule_directory_path is not None:
             schedule_directory = conf.inference.schedule_directory_path
         else:
-            schedule_directory = f"{SCRIPT_DIR}/../../schedules"
+            schedule_directory = os.getenv('RFDIFFUSION_SCHEDULES')
 
         # Check for cache schedule
         if not os.path.exists(schedule_directory):
@@ -148,7 +149,7 @@
         if self.inf_conf.input_pdb is None:
             # set default pdb
             script_dir=os.path.dirname(os.path.realpath(__file__))
-            self.inf_conf.input_pdb=os.path.join(script_dir, '../../examples/input_pdbs/1qys.pdb')
+            self.inf_conf.input_pdb=os.path.join(rfdiffusion_dir[0], './examples/input_pdbs/1qys.pdb')
         self.target_feats = iu.process_target(self.inf_conf.input_pdb, parse_hetatom=True, center=False)
         self.chain_idx = None
 
diff -ru RFdiffusion-1.1.0/scripts/run_inference.py RFdiffusion-1.1.0_data_paths/scripts/run_inference.py
--- RFdiffusion-1.1.0/scripts/run_inference.py	2023-04-03 23:33:05.000000000 +0200
+++ RFdiffusion-1.1.0_data_paths/scripts/run_inference.py	2023-08-08 18:06:05.112486398 +0200
@@ -28,6 +28,7 @@
 import random
 import glob
 
+from rfdiffusion import __path__ as rfdiffusion_dir 
 
 def make_deterministic(seed=0):
     torch.manual_seed(seed)
@@ -35,7 +36,7 @@
     random.seed(seed)
 
 
-@hydra.main(version_base=None, config_path="../config/inference", config_name="base")
+@hydra.main(version_base=None, config_path=os.path.join(rfdiffusion_dir[0], "config", "inference"), config_name="base")
 def main(conf: HydraConfig) -> None:
     log = logging.getLogger(__name__)
     if conf.inference.deterministic:
